src/Instrument/Transformer/FilterInjectorTransformer.php

```diff
@@ -97,7 +104,7 @@ public static function rewrite($originalResource, $originalDir = '')
             return self::PHP_FILTER_READ . self::$filterName . "/resource=" . $resource;
         }
 
-        $newResource = str_replace($appDir, $cacheDir, $resource);
+        $newResource = self::$kernel->getContainer()->get('aspect.cache.path.resolver')->getCachePathForResource($resource);
```

Comment:
Expensive method call, please, do not use container as service locator every time. This method will be called thousands of time, so this can be too slow. To optimize this, just store cache resolver instance directly.